src/language-mode.coffee

```diff
@@ -298,15 +319,30 @@ class LanguageMode
     indentLevel = @suggestedIndentForBufferRow(bufferRow, options)
     @editor.setIndentationForBufferRow(bufferRow, indentLevel, options)
 
+
+  # Count the number of occurence of regex in string, given its tokenized
+  # version, and ignoring any occurence that start inside a quoted string
+  countRegex: (regex, string, tokenized) ->
+    count = 0
+    m = regex.searchSync(string)
+    while (m)
+      quoted = false
+      if scopes = tokenized.tokenAtBufferColumn(m[0].start)?.scopes
+        quoted = scope for scope in scopes when scope.match '^string.quoted'
```

Comment:
Thanks for the suggestion of using `startsWith`, that is indeed what I was thinking.
However, regarding the loop, no, we cannot break it because the next occurrence may be outside of the quote again, e.g.,

```javascript
foo({
     a: 1
   /* an oddly placed comment */  });
var thisShouldBeAtIndent0;
```